runtime/calamari/src/lib.rs

```diff
@@ -131,70 +131,91 @@ pub const NORMAL_DISPATCH_RATIO: Perbill = Perbill::from_percent(70);
+    pub RuntimeBlockWeights: BlockWeights = BlockWeights::builder()
+        .base_block(BlockExecutionWeight::get())
+        .for_class(DispatchClass::all(), |weights| {
+            weights.base_extrinsic = ExtrinsicBaseWeight::get();
+        })
+        .for_class(DispatchClass::Normal, |weights| {
+            weights.max_total = Some(NORMAL_DISPATCH_RATIO * MAXIMUM_BLOCK_WEIGHT);
+        })
+        .for_class(DispatchClass::Operational, |weights| {
+            weights.max_total = Some(MAXIMUM_BLOCK_WEIGHT);
+            // Operational transactions have some extra reserved space, so that they
+            // are included even if block reached `MAXIMUM_BLOCK_WEIGHT`.
+            weights.reserved = Some(
+                MAXIMUM_BLOCK_WEIGHT - NORMAL_DISPATCH_RATIO * MAXIMUM_BLOCK_WEIGHT
+            );
+        })
+        .avg_block_initialization(AVERAGE_ON_INITIALIZE_RATIO)
+        .build_or_panic();
+    pub const SS58Prefix: u8 = manta_primitives::constants::CALAMARI_SS58PREFIX;
 }
 
 impl pallet_tx_pause::Config for Runtime {
-	type Event = Event;
-	type UpdateOrigin = EnsureOneOf<
-		EnsureRoot<AccountId>,
-		pallet_collective::EnsureProportionMoreThan<AccountId, TechnicalCollective, 1, 2>,
-	>;
-	type WeightInfo = weights::pallet_tx_pause::SubstrateWeight<Runtime>;
+    type Event = Event;
+    type UpdateOrigin = EnsureOneOf<
+        EnsureRoot<AccountId>,
+        pallet_collective::EnsureProportionMoreThan<AccountId, TechnicalCollective, 1, 2>,
+    >;
+    type WeightInfo = weights::pallet_tx_pause::SubstrateWeight<Runtime>;
 }
 
 // Don't allow permission-less asset creation.
 pub struct BaseFilter;
 impl Contains<Call> for BaseFilter {
-	fn contains(call: &Call) -> bool {
-		if matches!(
-			call,
-			Call::Timestamp(_) | Call::ParachainSystem(_) | Call::System(_)
-		) {
-			// always allow core call
-			// pallet-timestamp and parachainSystem could not be filtered because
-			// they are used in communication between relaychain and parachain.
-			return true;
-		}
-
-		if pallet_tx_pause::PausedTransactionFilter::<Runtime>::contains(call) {
-			// no paused call
-			return false;
-		}
-
-		match call {
-			| Call::Authorship(_)
-			| Call::Multisig(_)
+    fn contains(call: &Call) -> bool {
+        if matches!(
+            call,
+            Call::Timestamp(_) | Call::ParachainSystem(_) | Call::System(_)
+        ) {
+            // always allow core call
+            // pallet-timestamp and parachainSystem could not be filtered because
+            // they are used in communication between relaychain and parachain.
+            return true;
+        }
+
+        if pallet_tx_pause::PausedTransactionFilter::<Runtime>::contains(call) {
+            // no paused call
+            return false;
+        }
+
+        #[allow(clippy::match_like_matches_macro)]
+        // keep CallFilter with explicit true/false for documentation
+        match call {
+            // Explicitly DISALLOWED calls
```

Comment:
Previously we had disallowed calls listed in the `=> false` branch below the allowed ones.
If one adds a call into that branch while it's still present in the `=> true` branch, the `true` branch will take precedence and we'll have in-fact allowed calls we think are disallowed.

The false -> true fallthrough order makes sure that if we filter a call explicitly, it'll never be overwritten by for example a blanket allow of its pallet.

Reply:
that default fallthrough is still there at the bottom.

Reply:
I didn't get the CollatorSelection example. We'll have to maintain both whitelist and blacklist the way it's written here, no?
Or do you want to have a blacklist and allow everything else ?

Reply:
e.g. in this case:
```
// Disallowed
// Allowed
| Call::CollatorSelection(

// Disallowed
- => false
```
How do you filter ONLY `register_as_candidate` from that pallet.

Conversely:
```
// Disallowed
// Allowed
| Call::CollatorSelection(
				manta_collator_selection::Call::set_desired_candidates{..}
				| manta_collator_selection::Call::set_candidacy_bond{..}
				| manta_collator_selection::Call::set_eviction_baseline{..}
				| manta_collator_selection::Call::remove_collator{..}
// Disallowed
				| manta_collator_selection::Call::remove_collator{..}
				| - => false
```
if someone wants to ban remove_collator from a potentially long list of explicit extrinsics and forgets to remove it from above, allow takes precedence.

Reply:
> How do you filter ONLY register_as_candidate from that pallet.
That's the case in manta branch, just explicitly whitelist all except `register_as_candidate`

> if someone wants to ban remove_collator from a potentially long list of explicitly allowed extrinsics and forgets to remove it from above, allow takes precedence - silently.
We don't need an explicit list of disallowed that's my point, they should all just fall into the default case `| _ => false`
So in your example you would just remove it from the whitelist, no need to add it to the blacklist explicitly, so you can't make the mistake.

My other point is that even if we have a false, true, false arms I can still forget to move from one place to another, so there will be a silent failure

Reply:
I see two ways out of this:

a) we prioritize security against pallet upgrades adding new extrinsics and accidentally breaking stuff:
=> We list out _every single extrinsic_ we want to allow, no `Call:Scehduler(_)`, at which point we should move the filter into its own file

b) we prioritize compactness of expressing allowed/banned extrinsics
=> We keep my new approach.

I'm torn myself, we have the CI release script that detects metadata/extrinsic changes to hopefully prevent a), but us being invariant against parity pallets adding new extrinsics with some upgrade is enticing as well

Reply:
> > How do you filter ONLY register_as_candidate from that pallet.
> 
> That's the case in manta branch, just explicitly whitelist all except `register_as_candidate`

Ok, so either we roll this out across all runtimes **and all pallets**, which is gonna get large, (solution a from above) or

> We don't need an explicit list of disallowed that's my point, they should all just fall into the default case `| _ => false` So in your example you would just remove it from the whitelist, no need to add it to the blacklist explicitly, so you can't make the mistake.

But you still need to list all other extrinsics, not possible to collapse the case to `Call::CollatorSelection(_)` in that case, so it's *verbose*

Reply:
IMO the most secure is to have everything disallowed by default with the `| _ => false` and having to explicitly add "safe" calls to the allowed list.
a) what's wrong with `Call::Scheduler(_)` when I want all of them. At a point where we want to ban one or more of them we'd have to explicitly list the rest instead of `Call::Schedule(_)`.

Reply:
Right I see your point about this, better to explicitly write them out.

Reply:
Sure, let's log it and move this along. 
My vote would be for a) with the assumption that we will catch all new extrinsics, so we can keep using (_). At some point you just can't protect from dev mistakes : D